ONSdigital/census-rh-ui
# CR-1629 Exit Improvements

Signed-off-by: Paul-Joel <paul.joel@ons.gov.uk>

# Motivation and Context
Improvements to the authentication 'Exit' option

# What has changed
Added additional step for when 'Exit' is clicked in the Authentication journey to invalidate the session before redirecting back to the start page
Added new 'invalidate' method to clear the user session completely
Updated method for display of 'Exit' link to ensure it appears on all pages in the authentication journey except the timeout page
Added detection to unit tests to check that 'Exit' option is or isn't present depending on the journey

No Cucumber updates required

# How to test?
Submit a UAC, ensure that 'Exit' appears on subsequent pages, and that clicking 'Exit' results in the removal of the user session and a redirect to the 'start' page

# Links
https://collaborate2.ons.gov.uk/jira/browse/CR-1629

# Screenshots (if appropriate):
